runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -336,7 +356,9 @@ ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda,
 	UDATA newAnonClassNameLength = originalStringLength + 1 + ROM_ADDRESS_LENGTH + 1 + newHostPackageLength;
 
 	/* check if the class is a lambda class */
-	if (NULL != getLastDollarSignOfLambdaClassName(originalStringBytes, originalStringLength)) {
+	if (!isLambdaForm
+		&& (NULL != getLastDollarSignOfLambdaClassName(originalStringBytes, originalStringLength))
+	) {
```

Comment:
With this change, it is better to rename `isLambda` to sth like `isShareableLambda`.

Reply:
Lambda and LamdaForm are two different classes. They are not the same. Here, if a class is a LambdaForm, then we don't need to do the Lambda check.

Reply:
I think Hang is just talking from a code readability standpoint.

Reply:
What do we think here? I think leaving it as `isLambda` may just be better. But I'll leave that to consensus.

Reply:
I am fine with the logic here. `isShareableLambda` could be easier for other people to understand the code.

Reply:
I am fine with either.